common/src/main/java/io/netty/util/concurrent/AbstractEventExecutorGroup.java

```diff
@@ -46,22 +42,22 @@
     }
 
     @Override
-    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+    public Future<?> schedule(Runnable command, long delay, TimeUnit unit) {
```

Comment:
Should this and other methods taking a `Runnable` return `Future<Void>`?

Reply:
Yeah, `Void` is a better choice than wildcard.

Reply:
While I also thing that `Void` is better I wonder why the JDK did choose `ScheduledFuture<?>`. Any idea ?

Reply:
No, I don't know why they went with wildcards.

Reply:
Maybe something to ask for on the concurrency-interest mailing list just so we don't do something "stupid" ? That said this shouldn't block the PR

Reply:
Looks like it somehow is a, or ended up being a convention for futures that "return no value".

Reply:
hmm interesting.. Any pointers to this ?

Reply:
It's been like this from the start: http://gee.cs.oswego.edu/cgi-bin/viewcvs.cgi/jsr166/src/main/java/util/concurrent/ExecutorService.java?r1=1.7&r2=1.8
I think wildcard was chosen at the time because they specified the future to return "an arbitrary value".

I don't find any discussion of this from the beginning of the concurrency-interest list until 2004, when searching for all the relevant keywords I could come up with.

Reply:
Can you ask on the ml and link it here ? Just so we can decide later on what to do. For now let's just use `Void`.

Reply:
Asked: http://cs.oswego.edu/pipermail/concurrency-interest/2021-September/017420.html